ui/src/adapters/parsers/schemas.ts

```diff
@@ -1,349 +1,421 @@
 import { z } from "zod";
 
-import * as domain from "src/domain";
-import { StrictSchema } from "src/utils/types";
+import { getStrictParser } from "src/adapters/parsers";
```

Comment:
I feel some of the changes done in this PR are arbitrary. Specifically in this module, I think the only change I like is adding the `get` prefix to the functions that return a parser. For everything else, I'd revert the changes.

Let me reason a bit about what I don't like:

* In general in the PR you're moving all the types to the top of the module, while we discussed this in the past with the whole team and agreed to keep the types next to where they are used.
* In general in the PR you're adding groups like "// Parsers", "// Composite parsers" 🤯, "// Getters", "// Helpers", "// Exports", etc. IMO this categorization adds poor value and makes things more complex because one element can belong to 2 different categories. Not to mention that Parsers, Composite parsers, and even the Getters are the same to me: just parsers.
* I don't understand this "Composite" prefix you added to some types. Why add it here `type SchemaComposite = AnySchema & SchemaProps` and not here `type ArraySchema = CommonProps & ArrayPropsWithoutItems & { type: "array" }`?. Most types compose and that fact should never be part of its name.
* Same with the parsers. A parser is a parser and apart from the primitives, all parsers compose with others. The fact that we use `foo.and(bar)` to intersect two parsers does not qualify them to be tagged as "Composite parsers".
* Regarding the getters, I don't feel the need for the prefix, but I accept it because it makes more clear that they are function that expects some parameters required by the returned parser. This though does not mean we have to group them all under the "Getters" category.
* Qualifying `unsafelyParseMultiSchemaToDomain` as a helper is 🤯 to me. It's just another parser.
* As commented somewhere else, I'd not add an `// Exports` group since we can have exports in any other group. In this regard, I'd agree to remove all the export to an export statement at the end of the file.

As said, I'd revert all the changes in this module and then add the get prefixes to the functions.